# Belief-strategy consistency: evidence of manipulation

`belief_analysis.ipynb` established that in the treatment arm, a participant's `collabBelief` is shown to their partner via the robot's info modal (`"Partner Belief = " + partnerCollabBelief`), and that the *partner's* belief predicts a participant's own strategy choice specifically in treatment (the `partner_belief_c:arm` interaction). That creates an incentive that only exists in treatment: a participant could report an inflated `collabBelief` -- signalling they'll collaborate, to induce their partner into choosing `C` -- while actually planning to choose `I` themselves.

This is a viable exploit because of the payoff asymmetry documented throughout this analysis: `V_Y` (the individual design's payoff) doesn't depend on the partner's choice at all, so reporting a high belief and then defecting is never worse for the person doing it, and is only ever costly to their partner. This notebook asks: is there evidence of individual participants doing this systematically, more than the group as a whole?

**Operationalizing "inconsistent" -- as a continuous magnitude, not a threshold crossing.** This reuses the risk-dominance measure `u` from `risk_dominance_analysis.ipynb` -- for a given task, `u` is the minimum belief in the partner's cooperation needed to make choosing collaboratively the rational (higher expected-value) choice, given only *that participant's own* task payoffs:

$$u = \frac{V_Y^{II} - V_A^{CI}}{(V_Y^{II} - V_A^{CI}) + (V_A^{CC} - V_Y^{IC})}$$

For each (participant, round), `deviation = collabBelief - 100 * u_own`: how far the participant's stated belief sat above (positive) or below (negative) the threshold their own payoffs required to make `C` the rational choice. An earlier version of this analysis collapsed this to a binary `belief_favors_C` flag (`deviation >= 0`) and counted how often that co-occurred with choosing `I`. Collapsing to a threshold discards real information, though -- a participant whose belief barely crossed the line and a participant who wildly overstated it get treated identically, and it also can't distinguish "occasionally, mildly inconsistent" from "consistently defects despite claiming high confidence." Using the signed magnitude directly addresses this, and does so without needing to assume the same rational threshold applies equally to how confident someone *feels* -- individual differences in trust baseline or risk attitude shift where someone's true belief-choice boundary sits, but the size of `deviation` conditional on defecting is a more robust signal of the exploit pattern than crossing an assumed threshold at all.

`exploit_score` per round is `deviation` when the participant chose `I` and `deviation > 0` (i.e. their own payoff math implied `C` was better, but they chose `I` anyway), and `0` otherwise -- so it is zero for every round that isn't potentially exploitative, and scales with how much the stated belief overshot what would have been needed to justify the choice actually implied. Averaging `exploit_score` over **all** of a participant's rounds (not just their `I`-choice rounds) gives one continuous per-participant score that stays well-defined even for the many participants who never chose `I` at all (it's simply `0` for them), rather than requiring a person to have `I`-choice rounds to be scored.

Only `u_own` (the participant's own task) is used, not the round's paired `R` from the risk-dominance notebook -- `R` mixes in the *partner's* task difficulty, which the participant reporting the belief has no way to act strategically on for their own payoff calculation.

This is restricted to the **treatment arm only**: control has no belief-sharing channel, so there's no mechanism (or incentive) for this pattern to mean the same thing there.

**What this can and can't show.** A single inconsistent round doesn't prove intent -- it could be poor calibration, a change of mind, or noise. But the directional asymmetry (only *this* direction benefits the reporter at their partner's expense) plus testing each participant's *exploit score* against the group as a whole is a reasonable operationalization of "evidence consistent with manipulation," not proof of it for any individual round.

In [1]:
import numpy as np
import pandas as pd
from scipy import stats
from statsmodels.stats.multitest import multipletests

task = pd.read_csv("task_data.csv")
task_summary = pd.read_csv("task_summary.csv")

# task_summary.csv writes "n/a" for distraction/training tasks, which read_csv
# loads as NaN by default -- filter with notna(), not a string comparison
# (see risk_dominance_analysis.ipynb).
task_summary = task_summary[task_summary["task_difficulty"].notna()].copy()
task_summary["u"] = (task_summary["V_Y_II"] - task_summary["V_A_CI"]) / (
    (task_summary["V_Y_II"] - task_summary["V_A_CI"]) + (task_summary["V_A_CC"] - task_summary["V_Y_IC"])
)
u_by_index = task_summary.set_index("task_index")["u"]

## Drop missing data, restrict to treatment, reshape to one row per (participant, round)

Same drop criterion as the other outcome notebooks. Then restricted to `arm == "treatment"` for the reasons above, and reshaped exactly as in `belief_analysis.ipynb` -- one row per participant per round, since belief and strategy are individual, not joint, measures.

In [2]:
missing = (task["strategy_1"] == "undefined") | (task["strategy_2"] == "undefined")
task = task[~missing].copy()
task["pair_id"] = task["username_1"] + "_" + task["username_2"]

treatment = task[task["arm"] == "treatment"].copy()

shared_cols = ["pair_id", "round"]
belief = pd.concat([
    treatment[shared_cols + ["username_1", "collabBelief_1", "strategy_1", "task_1"]].rename(
        columns={"username_1": "username", "collabBelief_1": "collabBelief",
                 "strategy_1": "strategy", "task_1": "task"}),
    treatment[shared_cols + ["username_2", "collabBelief_2", "strategy_2", "task_2"]].rename(
        columns={"username_2": "username", "collabBelief_2": "collabBelief",
                 "strategy_2": "strategy", "task_2": "task"}),
], ignore_index=True)

belief["u_own"] = belief["task"].map(u_by_index)
belief["deviation"] = belief["collabBelief"] - belief["u_own"] * 100
belief["exploit_score"] = np.where((belief["strategy"] == "I") & (belief["deviation"] > 0), belief["deviation"], 0.0)
belief["generous_score"] = np.where((belief["strategy"] == "C") & (belief["deviation"] < 0), -belief["deviation"], 0.0)

print(f"n={len(belief)} (participant, round) observations, {belief['username'].nunique()} participants, "
      f"{belief['pair_id'].nunique()} pairs, all treatment")

n=840 (participant, round) observations, 28 participants, 14 pairs, all treatment


## Base rates

`generous_score` (the reverse pattern -- belief implied `I` was rational, but the participant chose `C` anyway) is included only for context, since the payoff asymmetry means it can't be an exploit of the partner in the same way; the rest of this notebook focuses on `exploit_score`.

In [3]:
n_exploit_rounds = (belief["exploit_score"] > 0).sum()
n_generous_rounds = (belief["generous_score"] > 0).sum()

print(f"exploit_score > 0:   {n_exploit_rounds} / {len(belief)} rounds ({n_exploit_rounds / len(belief):.1%}) "
      f"-- belief implied C was rational, chose I anyway")
print(f"  mean exploit_score, all rounds:      {belief['exploit_score'].mean():.2f} belief-points")
print(f"  mean exploit_score, exploit rounds only: {belief.loc[belief['exploit_score'] > 0, 'exploit_score'].mean():.2f}")

print(f"\ngenerous_score > 0:  {n_generous_rounds} / {len(belief)} rounds ({n_generous_rounds / len(belief):.1%}) "
      f"-- belief implied I was rational, chose C anyway")
print(f"  mean generous_score, all rounds:      {belief['generous_score'].mean():.2f} belief-points")
print(f"  mean generous_score, generous rounds only: {belief.loc[belief['generous_score'] > 0, 'generous_score'].mean():.2f}")

exploit_score > 0:   54 / 840 rounds (6.4%) -- belief implied C was rational, chose I anyway
  mean exploit_score, all rounds:      1.30 belief-points
  mean exploit_score, exploit rounds only: 20.23

generous_score > 0:  82 / 840 rounds (9.8%) -- belief implied I was rational, chose C anyway
  mean generous_score, all rounds:      1.23 belief-points
  mean generous_score, generous rounds only: 12.63


## Flagging individual participants

For each participant, `exploit_score` across **all 30 of their rounds** (not just their `I`-choice rounds) is compared against a **leave-one-out population baseline** (the pooled round-level `exploit_score` values of all *other* participants) with a one-sided Mann-Whitney U test -- a nonparametric rank-based test, appropriate here since `exploit_score` is heavily zero-inflated (most rounds score exactly `0`) and right-skewed on the rounds that don't, so a test relying on means and normality (like a t-test) would be a poor fit. With 28 participants tested, p-values are FDR-corrected (Benjamini-Hochberg) rather than read individually.

Using all 30 rounds (rather than only the `I`-choice subset) means every participant gets a score, including the many who never chose `I` at all -- their `exploit_score` is `0` for every round, which is the correct and informative value, not a missing one.

In [4]:
per_user = belief.groupby("username")["exploit_score"].agg(n_rounds="size", mean_exploit="mean", sum_exploit="sum")
per_user["n_positive"] = belief.groupby("username")["exploit_score"].apply(lambda s: (s > 0).sum())

p_values = []
for username in per_user.index:
    mine = belief.loc[belief["username"] == username, "exploit_score"]
    others = belief.loc[belief["username"] != username, "exploit_score"]
    result = stats.mannwhitneyu(mine, others, alternative="greater")
    p_values.append(result.pvalue)
per_user["p_value"] = p_values

_, per_user["p_adj_fdr"], _, _ = multipletests(per_user["p_value"], method="fdr_bh")
per_user["flagged"] = per_user["p_adj_fdr"] < 0.05

per_user.sort_values("p_value").round(4)

,n_rounds,mean_exploit,sum_exploit,n_positive,p_value,p_adj_fdr,flagged
username,,,,,,,
user0031,30,10.2728,308.1833,14,0.0000,0.0000,True
user0042,30,8.5865,257.5944,10,0.0000,0.0000,True
user0048,30,6.2532,187.5946,8,0.0000,0.0000,True
user0061,30,2.7303,81.9098,4,0.0577,0.3421,False
user0050,30,2.4681,74.0442,4,0.0611,0.3421,False
user0049,30,0.7462,22.3855,4,0.0803,0.3746,False
user0047,30,1.9888,59.6649,3,0.2074,0.7375,False
user0041,30,1.9223,57.6678,3,0.2107,0.7375,False
user0032,30,0.9624,28.8716,2,0.4968,0.9280,False


## Does a participant's exploit score relate to their pair's overall success?

A participant flagged here is, by construction, someone who often chose `I` when their own stated belief said `C` was clearly the better bet -- so their pair should tend to under-perform relative to what mutual collaboration would have achieved. This checks that directly: each participant's `mean_exploit` against their pair's overall successful-collaboration rate (`C`/`C` rounds, out of all 30 real-task rounds for that pair). Both Pearson and Spearman are reported -- Spearman is arguably the more appropriate summary here given how zero-inflated `mean_exploit` is (18 of 28 participants score exactly `0`).

In [5]:
treatment["success"] = ((treatment["strategy_1"] == "C") & (treatment["strategy_2"] == "C")).astype(int)
pair_success = treatment.groupby("pair_id")["success"].mean()

username_to_pair = pd.concat([
    treatment[["username_1", "pair_id"]].rename(columns={"username_1": "username"}),
    treatment[["username_2", "pair_id"]].rename(columns={"username_2": "username"}),
]).drop_duplicates().set_index("username")["pair_id"]

per_user["pair_id"] = per_user.index.map(username_to_pair)
per_user["pair_success_rate"] = per_user["pair_id"].map(pair_success)

r, p = stats.pearsonr(per_user["mean_exploit"], per_user["pair_success_rate"])
rs, ps = stats.spearmanr(per_user["mean_exploit"], per_user["pair_success_rate"])
print(f"Pearson r = {r:.3f}, p = {p:.4f} (n={len(per_user)} participants)")
print(f"Spearman r = {rs:.3f}, p = {ps:.4f}")

Pearson r = -0.685, p = 0.0001 (n=28 participants)
Spearman r = -0.941, p = 0.0000


For context, per-pair robot-use rate alongside success rate (referenced in the interpretation below, to check whether the flagged participants' pairs also stand out on robot use, as found independently in `robot_use_analysis.ipynb`).

In [6]:
treatment["either_used_robot"] = (treatment["usedRobot_1"] | treatment["usedRobot_2"]).astype(int)
per_pair_context = treatment.groupby("pair_id").agg(
    robot_use_rate=("either_used_robot", "mean"), success_rate=("success", "mean"),
)
per_pair_context.sort_values("robot_use_rate", ascending=False).round(3)

,robot_use_rate,success_rate
pair_id,,
user0017_user0018,1.000,1.000
user0041_user0042,1.000,0.400
user0047_user0048,0.733,0.333
user0061_user0062,0.633,0.800
user0033_user0034,0.533,1.000
user0049_user0050,0.467,0.300
user0015_user0016,0.233,1.000
user0055_user0056,0.200,1.000
user0057_user0058,0.167,1.000


## Interpretation

**Base rates.** 6.4% of treatment (participant, round) observations (54 of 840) had a positive `exploit_score` -- the participant's own payoff math implied `C` was rational, yet they chose `I` anyway. Conditional on that happening, the average overshoot was substantial (mean `exploit_score` of 20.2 belief-points among those 54 rounds). For context, the reverse pattern is actually more common: 9.8% of rounds (82 of 840) had a positive `generous_score`, and participants defaulted toward collaborating more often than their own stated belief implied was rational -- the opposite of a manipulative pattern.

**Individual flagging.** Of 28 treatment participants, three survive FDR correction as significantly more exploitative than the leave-one-out population baseline (one-sided Mann-Whitney U on round-level `exploit_score`): `user0031` (mean 10.27, 14 of 30 rounds with a positive score), `user0042` (mean 8.59, 10 of 30), and `user0048` (mean 6.25, 8 of 30). Three more sit at an elevated but not-significant score after correction -- `user0061` and `user0050` (p_adj ≈ 0.34) and `user0049` (p_adj ≈ 0.37) -- worth noting as a second tier rather than treating as confirmed. 18 of the 28 participants score exactly `0` on every round; the behavior is concentrated in a small minority, not spread across the sample.

This flags the same three participants as an earlier binary version of this analysis (which counted, rather than weighted by magnitude, rounds where a `belief_favors_C` threshold was crossed alongside choosing `I`) -- but the ranking among them is now magnitude-sensitive (`user0031` clearly exceeds `user0042` and `user0048`, not just tied at "flagged"), and the earlier version's borderline flag for `user0048` is now much more clearly separated from the pack rather than resting on a handful of barely-over-the-line rounds.

**Corroboration.** A participant's `mean_exploit` correlates strongly with their pair's overall successful-collaboration rate (Pearson r = -0.685, p < 0.001; Spearman r = -0.941, p < 0.001, n = 28 -- Spearman is the more appropriate summary given how zero-inflated the score is). Part of this is close to definitional (an `I` choice can't itself be part of a `C`/`C` round, whether or not it was "inconsistent"), so this alone isn't independent proof of manipulation. What *is* independent, and more telling: the three flagged participants line up by name with pairs already flagged as anomalous elsewhere in this analysis, using entirely different data. `user0031` is from the `user0031_user0032` pair singled out in `outcome_analysis.ipynb`'s robustness check for behaving like a control pair despite being assigned to treatment, and confirmed here as one of only two treatment pairs with a 0% robot-use rate. `user0042` and `user0048` are from `user0041_user0042` and `user0047_user0048` -- the pairs with the highest (100%) and third-highest (73.3%) robot-use rates in the whole treatment arm, and two of its three lowest success rates (40.0% and 33.3%, against a treatment median of 100%). Three separate analyses, using separate variables (outcome rates, robot-use frequency, belief-strategy consistency), converge on the same small set of pairs -- that convergence is the strongest evidence here, stronger than any single measure alone.

**Caveats.** This flags *patterns* consistent with manipulation, not intent -- `u_own` assumes a stylized expected-value-maximizing model of what's "rational," and a participant may not reason about their own payoffs this precisely. The Mann-Whitney test also treats each participant's rounds as independent observations, which likely understates the true variability of a serially-correlated behavior. Given all that, this is best read as a data-driven shortlist for closer qualitative review (e.g. by round-by-round timeline) rather than a final determination of who manipulated whom.